pyflwdir/flwdir.py

```diff
@@ -81,10 +81,12 @@ def __init__(
         self.idxs_outlet = idxs_outlet
         self._seq = idxs_seq
         self._nnodes = nnodes
-        # either -1 for int or 4294967295 for uint32
+        # either -1 for int, 4294967295 for uint32, or 18446744073709551615 for uint64
         self._mv = core._mv
         if idxs_ds.dtype == np.uint32:
             self._mv = np.uint32(self._mv)
+        if idxs_ds.dtype == np.uint64:
```

Comment:
Just appears to be a dtype limitation?

Reply:
@robgpita

Reply:
@fernando-aristizabal not exactly.  The dtype assigned to `self._mv` needs to be the same as the dtype for `idxs.ds`. Those two dtypes need to line up, otherwise there were errors in numba code.  In the [downstream code](https://github.com/Deltares/pyflwdir/blob/45c5e63615a9919af9c7ea9719c3dde4f975d425/pyflwdir/core.py#L12),  incorrect aliasing was occurring based on `_mv`'s dtype. This was causing a mis-match in values between  `_mv` and items in the `idxs.ds` array, which ultimately led to the IndexErrors being thrown. 

In order to avoid the IndexErrors, for the large array indices  case, where the `idxs.ds.dtype` is [set to uint64](https://github.com/Deltares/pyflwdir/blob/45c5e63615a9919af9c7ea9719c3dde4f975d425/pyflwdir/pyflwdir.py#L167), the dtype for `_mv` needs to be set to uint64 as well.